autosklearn/util/progress_bar.py

```diff
@@ -0,0 +1,38 @@
+import time
+
+from threading import Thread
+from tqdm import trange
+
+
+class ProgressBar(Thread):
+    """A Thread that displays a tqdm progress bar in the console."""
+
+    def __init__(self, total: float, update_interval: float = 1.0, disable: bool = False):
+        """
+        Parameters
+        ----------
+        total: the total amount that the progress bar should reach
+        update_interval: reduce this to update the progress bar more frequently
+        disable: flag that turns on or off the progress bar. If false, then no thread is started or created.
+        """
+        self.disable = disable
+        if not disable:
+            super().__init__(name="_progressbar_")
+            self.total = total
+            self.update_interval = update_interval
+            self.terminated: bool = False
+            self.start()
+
+    def run(self):
```

Comment:
Mypy will warn you but you can specify the return type as such. Mypy has an option that no return type is implicitly None but I'd rather be explicit about it.

```python
def run(self) -> None:
    ...
```